### PR TITLE
hotfix: extractCSS error in dev mode

### LINKED ISSUE #4885: Setting `extractCSS` to `true` cause error: `Invalid CSS after "...  var cssReload": expected selector, was '= require`
### Version

[v2.4.0](https://github.com/nuxt.js/releases/tag/v2.4.0)

### Reproduction link

[https://codesandbox.io/s/z6510o74ql?autoresize=1&hidenavigation=1&module=%2Fnuxt.config.js&moduleview=1](https://codesandbox.io/s/z6510o74ql?autoresize=1&hidenavigation=1&module=%2Fnuxt.config.js&moduleview=1)

### Steps to reproduce

- Create a new Nuxt@^2.4.0
- Add `SCSS` support `npm install --save-dev node-sass sass-loader`
- Set `extractCSS` to `true` in the `build` property of `nuxt.config.js`
- Set an external 'scss' file for 'index.vue' page

### What is expected ?

To work as expected

### What is actually happening?

Error in console and in browser:

Module build failed (from ./node_modules/extract-css-chunks-webpack-plugin/dist/loader.js):
ModuleBuildError: Module build failed (from ./node_modules/sass-loader/lib/loader.js):

undefined
                   ^
      Invalid CSS after "...  var cssReload": expected selector, was '= require("../node_'

### Additional comments?

- 2.3.4 was working ok
- Seems a problem related to 'SCSS' loader.

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c8564">#c8564</a>)</em></sub></div>

### PR DESCRIPTION
**Pull request by clarkdo**
## Types of changes

- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

Fix #4885.
Related #4888 

As a hotfix, we've already disabled extractCSS in dev mode, this ps is a better fix for solving the root error, but for stability, we may test more and discuss about if there is any value for enabling extractCSS in dev/HMR mode

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [ ] All new and existing tests are passing.

### CODE REVIEW

**Review by pi0 — APPROVED** [review]

File: packages/webpack/src/config/base.js [hunk 1]
```diff
@@ -327,15 +327,27 @@ export default class WebpackBaseConfig {
...
+        filename: this.getFileName('css'),
+        chunkFilename: this.getFileName('css'),
+        // TODO: https://github.com/faceyspacey/extract-css-chunks-webpack-plugin/issues/132
+        reloadAll: true
+      }, this.options.build.extractCSS)))
+    }
+
+    plugins.push(new VueLoader.VueLoaderPlugin())
+
+    Array.prototype.push.apply(plugins, this.options.build.plugins || [])
```
Comment (manniL) [inline packages/webpack/src/config/base.js]: @clarkdo do you know why we can't use `plugins.push` here?

Comment (pi0) [reply packages/webpack/src/config/base.js]: It was a legacy issue :laughing: I think it is now even safe to use rest operator.

Comment (clarkdo) [reply packages/webpack/src/config/base.js]: @manniL No special reason, it's old code, I didn't notice it

**Comment by clarkdo** [comment]
@pi0 @manniL How do you think below code ? I think it's less readable and not perform as good as push.

```js

  plugins() {
    return [
      ...(this.options.dev ? new TimeFixPlugin() : []),
      ...this.options.build.extractCSS
        ? new ExtractCssChunksPlugin(Object.assign({
          filename: this.getFileName('css'),
          chunkFilename: this.getFileName('css'),
          // TODO: https://github.com/faceyspacey/extract-css-chunks-webpack-plugin/issues/132
          reloadAll: true
        }, this.options.build.extractCSS))
        : [],
      new VueLoader.VueLoaderPlugin(),
      ...(this.options.build.plugins || []),
      // Hide warnings about plugins without a default export (#1179)
      new WarnFixPlugin(),
      new WebpackBar({
        name: this.name,
        color: this.colors[this.name],
        reporters: [
          'basic',
          'fancy',
          'profile',
          'stats'
        ],
        basic: !this.options.build.quiet && env.minimalCLI,
        fancy: !this.options.build.quiet && !env.minimalCLI,
        profile: !this.options.build.quiet && this.options.build.profile,
        stats: !this.options.build.quiet && !this.options.dev && this.options.build.stats,
        reporter: {
          change: (_, { shortPath }) => {
            if (!this.isServer) {
              this.nuxt.callHook('bundler:change', shortPath)
            }
          },
          done: (context) => {
            if (context.hasErrors) {
              this.nuxt.callHook('bundler:error')
            }
          },
          allDone: () => {
            this.nuxt.callHook('bundler:done')
          }
        }
      }),
      ...this.options.build.hardSource
        ? new HardSourcePlugin(Object.assign({}, this.options.build.hardSource))
        : []
    ]
  }
```

**Comment by manniL** [comment]
Agree with @pi0 :relaxed:

**Pull request merged**